addons/mail/static/tests/tours/discuss_channel_public_tour.js

```diff
@@ -55,9 +55,36 @@ registry.category("web_tour.tours").add("discuss_channel_public_tour.js", {
             trigger: ".o-mail-AttachmentCard:not(.o-isUploading)", // waiting the attachment to be uploaded
         },
         {
-            content: "Check the earlier provided attachment is listed",
+            content: "Check the text attachment is listed",
             trigger: '.o-mail-AttachmentCard[title="text.txt"]',
         },
+        {
+            content: "Add an image file in composer",
+            trigger: ".o-mail-Composer button[aria-label='Attach files']",
+            async run() {
+                await inputFiles(".o-mail-Composer-coreMain .o_input_file", [
+                    await createFile({
+                        contentType: "image/jpeg",
+                        name: "image.jpg",
+                    }),
+                ]);
+            },
+        },
+        {
+            trigger: ".o-mail-AttachmentImage:not(.o-isUploading)",
+        },
+        {
+            content: "Check the image attachment is listed",
+            trigger: '.o-mail-AttachmentImage[title="image.jpg"]',
+            async run() {
+                const token = Object.values(
+                    odoo.__WOWL_DEBUG__.root.env.services["mail.store"].Attachment.records
+                ).filter((a) => a.name === "image.jpg")[0].access_token;
```

Comment:
This assert might be a little weak:
It will crash if any other existing `image.jpg` attachment exists and is known when running the test.
I think it would be a lot more robust to use `.at(-1)` to get the last attachment (which should always be the one just created) rather than `[0]`. The filter on name can stay for extra guarantees.

Reply:
Or maybe we can parse `src` to find the `id` and then look for the correct attachment directly?